### ISSUE TITLE
chart.update() refression with yAxis

### ISSUE THREAD

**Issue by mermonkey** [highcharts/highcharts-react#465]
hi! This use-case is a little complicated and I didn't write it, so I'll try my best to describe the issue. 
We are using highcharts-react inside a render function. There is some parent data that switches between different charts (by updating props causing the render function to return a new <HighchartsReact> element. This worked fine with v9.0.1 of highCharts and v3.0.0 of highcharts-react. Recently upgraded to v11.3 of highCharts and v3.2.1 of hc-react and seeing a problem. First chart comes up ok, but when we switch to a different chart, the y-axis grid lines / ticks are mostly lost (only first and last are present). Noticed that if i pass an empty yAxis object to hc, we never have a problem (options.yAxis={}). Only thing being used on yAxis is title.text and title.text.color; everything else should be default (e.g., tickPixelInterval, etc). Again, this works on the first chart, but fails on the subsequent charts. hc-react is using chart.upate() in the non-working case and the working case with no yAxis config.
Also, setting immutable=true works but of course uses newChart() instead of chart.update().
Any idea what could be going wrong here?
I'm tempted to always use immutable since I'm not sure we have any/many use-cases that benefit from the perf gains of update.
Thanks!

**Comment by ppotaczek** [comment]
Hi @mermonkey,

Thank you for contacting us!

I guess that some default options for yAxis (which can be different for different chart types) are stored in the chart object and used after an update.

Could you reproduce the problem in some online code editor? You can use this demo as a base to start: https://stackblitz.com/edit/react-fmsgks

Best regards!

**Comment by mermonkey** [comment]
Thanks for the reply @ppotaczek 
I'll look at recreating this on stackblitz. Not sure what is different about our usage atm.
Is there something on the chart object I could reset to force the yAxis to regenerate the ticks/gridlines? Setting options.yAxis={} works, but I'd like to set yAxis.title...
I tried this: chart.yAxis[0] = {} but seems too aggressive. Highcharts complains TypeError; Cannot read properties of undefined (reading 'isInternal').

**Comment by mermonkey** [comment]
@ppotaczek I'm not able to reproduce on stackblitz so far.
Any ideas on work-arounds I could try?
If I don't set custom yAxis text, it works fine...

**Comment by KamilKubik** [comment]
Hello @mermonkey!

Unfortunately, without a reproduction version, it is difficult to help, as we cannot debug the problem. It would be much more helpful if you were able to reproduce it.

As mentioned by @ppotaczek, different chart types with different options may be related. May I ask what chart types you switch to when there is an issue? 

Also, what should be helpful to reproduce it would mainly be the `series` options/configurations before and after switching to the other chart. Could you provide the other chart options used (apart from the `yAxis` property)?

Looking forward to your response,
Kind Regards!

**Comment by mermonkey** [comment]
Thanks @KamilKubik 
It's a scatter chart. 
This is what is set on the options.yAxis:
            "title": {
                "text": "Z(m)",
                "style": {
                    "color": "rgb(70, 70, 70)"
                }
            }
     
If we don't set any yAxis options, chart.update is ok.

**Comment by KamilKubik** [comment]
Thanks for sharing this but unfortunately, I still have insufficient information to reproduce it: [demo](https://stackblitz.com/edit/react-d9vtql?file=index.js,ChartComponent.js).

As I mentioned, what would be helpful is all the options you're passing to the charts provided in two states, before and after an update, since several of them may be related combined with the `yAxis` property, not the `yAxis` itself. You can update the above demo with these options and the way the rerender is fired (whether the chart options are updated during re-rendering, or perhaps it's caused by the unrelated state).

According to your mention: `I tried this: chart.yAxis[0] = {} but seems too aggressive`, you can try to fire the `update` method manually on a chart's instance instead. You can refer to it as described [here](https://github.com/highcharts/highcharts-react?tab=readme-ov-file#how-to-get-a-chart-instance) or by referring to the global `Highcharts.charts` array (and the specific chart), and then, use the [update](https://api.highcharts.com/class-reference/Highcharts.Chart#update) method. 

Regards!

**Comment by mermonkey** [comment]
Thanks @KamilKubik 
I updated your example with some similar before/after options here: https://react-9i8mtk.stackblitz.io 
I'm noticing though that when I click the "update chart" button, newChart() is used instead of chart.update(). How can we update this example to use chart.update(). Is there a problem with "renderTo"? 
thanks.

**Comment by KamilKubik** [comment]
Thanks for sharing the updated demo but unfortunately I can't open it (I tried on different machines but it can't load - probably StackBlitz faces some issues).

Could you send me an `Editor URL` link instead of the `Application URL`? Alternatively, you can fork the demo, and copy the URL.

Regards!

**Comment by mermonkey** [comment]
hi @KamilKubik ,
Does this work now? https://stackblitz.com/edit/react-9i8mtk?file=index.js

**Comment by mermonkey** [comment]
My original issue is not visible in your new example using update() directly. Not sure what is different.
This is how the orig issue looks:

![image](https://github.com/highcharts/highcharts-react/assets/3648298/65be6224-fe26-4682-b543-b95a0289a1b3)

I'm possibly not clear on all the suggestions. As mentioned, setting immutable=true does fix my problem but i'd like to avoid it for performance reasons. Is there another suggestion I should try?

**Comment by ppotaczek** [comment]
Hi @mermonkey, I was on vacation, so Kamil took over the topic for a moment, but I'm coming back to you

> I'm noticing though that when I click the "update chart" button, newChart() is used instead of chart.update(). How can we update this example to use chart.update(). Is there a problem with "renderTo"?

The wrapper creates its own container for a chart, so the `renderTo` option is ignored. However, could you clarify why you think that a new chart is created instead of updating the existing one? In this example: https://stackblitz.com/edit/react-e5hodk I've added `redraw` event callback which is fired on `chart.update` and not on creating a new chart.

API: 
https://api.highcharts.com/highcharts/chart.renderTo
https://api.highcharts.com/highcharts/chart.events.redraw

I see you still haven't been able to reproduce the issue in the code editor and also I don't see anything wrong in your code, so it's really hard to guess what is wrong. Maybe you could try to add more logic and component structure from the original use case? Maybe the code editor environment is a limitation and you could try to reproduce the problem in some separate minimal react project and provide it here?

Also, there is no direct option to force the yAxis to regenerate the ticks. You can try to set [tickAmount](https://api.highcharts.com/highcharts/yAxis.tickAmount), but the built-in logic for that will be disabled.

Best regards!

**Comment by mermonkey** [comment]
I think new chart is created because createChart() on line 54 of HighchartsReact.js is executed. 
In my original case, chartRef.current.update() is called from line 67 instead. 
Possibly I could demo the original case to you in a quick debug session?

**Comment by ppotaczek** [comment]
How did you check that? Maybe some React states don't work correctly in your debugging environment? Here: https://stackblitz.com/edit/react-qlpf6l I just used `HighchartsReact` as a basic component and the `createChart` and `update` functions are called in the right order.

Sorry, but we don't provide support on live debug sessions.

**Comment by mermonkey** [comment]
I checked that by setting breakpoints in chrome dev inspector. Possibly src map is imperfect or something. In your example, i see update being called as expected. Kind of stuck now...
In summary:
- this worked in v9; regressed when we moved to v11
- works if i don't set a yAxis.title on update
- works if i force a new chart with immutable=true
Any other ideas?

**Comment by ppotaczek** [comment]
Yes, there must be something wrong with the dev inspector. In other case enabling the immutable option would have no effect and the `chart.redraw` event wouldn't be fired.

As I said before, my debugging possibilities are limited due to a lack of problem reproduction. However, you can try the things below:

- check if updating only series and yAxis also causes the issue (like [here](https://stackblitz.com/edit/react-mt6p3q?file=index.js))
- use Highcharts source code and debug why only two yAxis ticks are added (for example by wrapping the `setTickPositions` method like [here](https://stackblitz.com/edit/react-gjcww7?file=index.js))

**Comment by ppotaczek** [comment]
Hi @mermonkey,
The problem seems to be related with the boost module. Another user reported similar problem here: https://github.com/highcharts/highcharts/issues/21068

**Comment by mermonkey** [comment]
@ppotaczek you are correct!
When I stop using boost, the problem does not occur.
What's next?
Thanks!

**Comment by ppotaczek** [comment]
The issue is marked as a regression bug, but there is no ETA for fixing it. You can add a comment in the thread indicating that you have the same problem.

As a workaround, you can use Highcharts v11.3.

**Comment by mermonkey** [comment]
thanks. I am seeing this with 11.3 though...
@ppotaczek

**Comment by ppotaczek** [comment]
@mermonkey,
Sorry, my mistake. As a workaround, use Highcharts v11.2. 
Example in pure JS: https://jsfiddle.net/BlackLabel/5Lehxfzm/

Best regards!
